Website/src/rendering/src/components/PreviewSearch/PreviewSearch.tsx

```diff
@@ -45,24 +44,19 @@ const PreviewSearch = ({
 
   const changeCategory = debounce(
     (categoryUrl: string) => {
-      const category = getCategoryByUrlPath(categoryUrl);
-      if (!category) {
-        return;
+      if (isDiscoverEnabled) {
+        const category = getCategoryByUrlPath(categoryUrl);
+        if (!category) {
+          return;
+        }
+        // HACK: Clear the keyphrase before changing the category to display all the products of that category
+        onKeyphraseChange({ keyphrase: '', rfkId: rfkId });
+        onCategoryChange({ category: category.ccid, rfkId: rfkId });
```

Comment:
With Discover enabled, hovering on a trending/top category does not refresh the results anymore. I suspect the same for "did you mean" suggestions.

Reply:
This is the same behavior I am observing [on the deployed site](https://edge-shop-website.sitecoredemo.com) at least for trending categories. I'm wondering if the "did you mean" suggestion is related to the caching issue called out [here](https://github.com/Sitecore/Sitecore.Demo.Edge/pull/449#discussion_r953126509)

Reply:
The "did you mean" suggestions should work now (related to caching issue)

trending/top category does not refresh results, and I see [this comment](https://github.com/Sitecore/Sitecore.Demo.Edge/blob/develop/Website/src/rendering/src/components/PreviewSearch/PreviewSearch.tsx#L78) by Christos documenting that behavior/bug I believe this is possibly an issue with the discover sdk